File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        dct = {")": "(", "}": "{", "]": "["}

        stack = []
        for c in s:
            if c in dct:
                if stack and stack[-1] == dct[c]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(c)

        return True if not stack else False

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;

        for (string& s : tokens) {
            if (s == "+"){
                int num2 = st.top(); st.pop();
                int num1 = st.top(); st.pop();
                st.push(num1 + num2);
            }
            else if (s == "-"){
                int num2 = st.top(); st.pop();
                int num1 = st.top(); st.pop();
                st.push(num1 - num2);
            }
            else if (s == "*"){
                int num2 = st.top(); st.pop();
                int num1 = st.top(); st.pop();
                st.push(num1 * num2);
            }
            else if (s == "/"){
                int num2 = st.top(); st.pop();
                int num1 = st.top(); st.pop();
                st.push(num1 / num2);
            }
            else{
                st.push(stoi(s));
            }
        }

        return st.top();       
    }
};
